### PR TITLE
[SPARK-41897][CONNECT][TESTS] Enable tests with error mismatch in connect/test_parity_functions.py

### PR DESCRIPTION
**Pull request by techaddict**
### What changes were proposed in this pull request?
Fix tests with error mismatch in connect/test_parity_functions.py

### Why are the changes needed?
Tests coverage

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Enabling new tests

### CODE REVIEW

File: python/pyspark/sql/tests/test_functions.py [hunk 1]
```diff
@@ -964,15 +965,11 @@ def test_raise_error(self):
...
         df = self.spark.createDataFrame([Row(id="foobar")])
 
-        with self.assertRaises(Py4JJavaError) as cm:
+        with self.assertRaisesRegex((Py4JJavaError, SparkConnectException), "foobar") as cm:
             df.select(raise_error(df.id)).collect()
-        self.assertIn("java.lang.RuntimeException", str(cm.exception))
-        self.assertIn("foobar", str(cm.exception))
 
-        with self.assertRaises(Py4JJavaError) as cm:
+        with self.assertRaisesRegex((Py4JJavaError, SparkConnectException), "barfoo") as cm:
```
Comment (HyukjinKwon) [inline python/pyspark/sql/tests/test_functions.py:971]: This is nice. We should fix other tests like this.

**Review by HyukjinKwon — DISMISSED** [review]

File: python/pyspark/sql/tests/test_functions.py [hunk 2]
```diff
@@ -24,6 +24,7 @@
 
 from py4j.protocol import Py4JJavaError
 from pyspark.sql import Row, Window, types
+from pyspark.sql.connect.client import SparkConnectException
```
Comment (HyukjinKwon) [inline python/pyspark/sql/tests/test_functions.py:27]: Ah, importing this requires `pyarrow` to be installed. The errors are being refactored in https://github.com/apache/spark/pull/39387. We should probably take these exceptions out to a separate package so it doesn't require to import other libraries.

Comment (techaddict) [reply python/pyspark/sql/tests/test_functions.py:27]: @HyukjinKwon ya makes sense, will update this once #39387 is merged 👍🏼

Comment (HyukjinKwon) [reply python/pyspark/sql/tests/test_functions.py:27]: https://github.com/apache/spark/pull/39387 is merged. Could we move the Spark Connect errors into there, see fix the test cases here? See also https://github.com/apache/spark/pull/39591. cc @itholic

**Comment by techaddict** [comment]
@HyukjinKwon updated the PR

**Review by HyukjinKwon — APPROVED** [review]
Thanks for fixing it.

File: python/pyspark/sql/tests/test_functions.py [hunk 3]
```diff
@@ -1015,48 +1015,42 @@ def test_assert_true(self):
...
-        self.assertIn("too big", str(cm.exception))
 
-        with self.assertRaises(Py4JJavaError) as cm:
+        with self.assertRaisesRegex((Py4JJavaError, SparkConnectException), "2000000"):
             df.select(assert_true(df.id < 2, df.id * 1e6)).toDF("val").collect()
-        self.assertIn("java.lang.RuntimeException", str(cm.exception))
-        self.assertIn("2000000", str(cm.exception))
 
-        with self.assertRaises(PySparkTypeError) as pe:
+        with self.assertRaises((PySparkTypeError, TypeError)) as pe:
```
Comment (HyukjinKwon) [inline python/pyspark/sql/tests/test_functions.py:1024]: @itholic I think we should fix this not to throw `TypeError` too for Spark Connect.

**Review by zhengruifeng — APPROVED** [review]

File: python/pyspark/sql/tests/test_functions.py (hunk 3 above)
Comment (itholic) [reply python/pyspark/sql/tests/test_functions.py:1024]: Yeah, I think so. Let me submit a fix soon.

**Review by itholic — APPROVED** [review]
Looks good. Will submit a PR for migrating Python built-in errors in Spark Connect to PySpark error framework and refine the tests as follow-ups.

**Comment by HyukjinKwon** [comment]
Merged to master.

File: python/pyspark/sql/tests/test_functions.py (hunk 3 above)
Comment (itholic) [reply python/pyspark/sql/tests/test_functions.py:1024]: FYI: Submit https://github.com/apache/spark/pull/39782 to fix this.

**Pull request closed without merging**